File: src/main/java/com/challenge/interview/calendar/api/config/WebConfig.java
package com.challenge.interview.calendar.api.config;

import com.challenge.interview.calendar.model.RoleStringToEnumConverter;
import lombok.NonNull;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.convert.converter.Converter;
import org.springframework.format.FormatterRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class WebConfig implements WebMvcConfigurer {
    public void addFormatters(@NonNull FormatterRegistry registry) {
        registry.addConverter((Converter)(new RoleStringToEnumConverter()));
    }
}

File: src/main/kotlin/com/challenge/interview/calendar/api/config/WebConfig.kt
package com.challenge.interview.calendar.api.config

import com.challenge.interview.calendar.model.StringToEnumConverter
import org.springframework.context.annotation.Configuration
import org.springframework.format.FormatterRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebConfig : WebMvcConfigurer {
    override fun addFormatters(registry: FormatterRegistry) {
        registry.addConverter(StringToEnumConverter())
    }
}